src/test/anovos/feature_recommender/test_feature_recommendation.py

```diff
@@ -0,0 +1,231 @@
+    assert test_feature_corpus[1] == list(dict.fromkeys(test_order_unique_list))[1]
+    for i in range(len(test_df)):
+        assert "N/A" in test_df.iloc[i, 3] or float(test_df.iloc[i, 3]) >= 0.3
+        assert float(test_df.iloc[i, 3]) <= 1
+
+    test_df_2 = find_attr_by_relevance(
+        example_attr_2,
+        building_corpus=test_feature_corpus_2,
+        desc_column="Desc",
+        threshold=0.25,
+    )
+    assert len(test_df_2) > 0
+    assert "Recommended_Input_Attribute_Name" not in test_df_2.columns
+    assert "Recommended_Input_Attribute_Description" in test_df_2.columns
+    assert "Input_Feature_Description" in test_df_2.columns
+    assert "Input_Attribute_Similarity_Score" in test_df_2.columns
+    for i in range(len(test_df)):
+        assert "N/A" in test_df.iloc[i, 3] or float(test_df.iloc[i, 3]) >= 0.25
+        assert float(test_df.iloc[i, 3]) <= 1
+
+    test_df_3 = find_attr_by_relevance(
+        example_attr_1,
+        building_corpus=test_feature_corpus_3,
+        name_column="Attribute Name",
+    )
+    assert len(test_df_3) > 0
+    assert "Recommended_Input_Attribute_Name" in test_df_3.columns
+    assert "Recommended_Input_Attribute_Description" not in test_df_3.columns
+    assert "Input_Feature_Description" in test_df_3.columns
+    assert "Input_Attribute_Similarity_Score" in test_df_3.columns
+    test_order_unique_list_3 = test_df_3.iloc[:, 0].to_list()
+    assert test_feature_corpus_3[0] == list(dict.fromkeys(test_order_unique_list_3))[0]
+    assert test_feature_corpus_3[1] == list(dict.fromkeys(test_order_unique_list_3))[1]
+    for i in range(len(test_df_3)):
+        if "N/A" in test_df_3.iloc[i, 2]:
+            assert "N/A" in test_df_3.iloc[i, 1]
+
+
+def test_sankey_visualization(example_attr_1, example_attr_2):
+    test_feature_corpus = [
+        "number of customers using products",
+        "number of call customer make daily",
+    ]
+    test_df = feature_recommendation(
+        example_attr_1,
+        name_column="Attribute Name",
+        desc_column="Attribute Description",
+    )
+    test_df_2 = feature_recommendation(
+        example_attr_2,
+        desc_column="Desc",
+        suggested_industry="banking",
+        suggested_usecase="credit risk",
+    )
+    test_df_3 = find_attr_by_relevance(
+        example_attr_1,
+        building_corpus=test_feature_corpus,
+        name_column="Attribute Name",
+        desc_column="Attribute Description",
+    )
+
+    sankey_plot = sankey_visualization(test_df)
+    assert "churn" in sankey_plot.data[0]["node"]["label"]
+    assert "AccountWeeks" in sankey_plot.data[0]["node"]["label"]
+    assert "telecommunication" not in sankey_plot.data[0]["node"]["label"]
+    assert (
+        "banking financial service and insurance"
+        not in sankey_plot.data[0]["node"]["label"]
+    )
+
+    sankey_plot_2 = sankey_visualization(
+        test_df_2, industry_included=True, usecase_included=True
+    )
```

Comment:
Is there something we can check to confirm on this method call ?

Reply:
For this, the type of the variable will be plot, and there are elements inside that can only be accessed while the object is plot.